src/tmate.py

```diff
@@ -138,6 +141,32 @@ def install_keys(host_ip: typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Addr
         raise KeyInstallError from exc
 
 
+def _wait_for(
+    func: typing.Callable[[], typing.Any], timeout: int = 300, check_interval: int = 10
+) -> None:
+    """Wait for function execution to become truthy.
+
+    Args:
+        func: A callback function to wait to return a truthy value.
+        timeout: Time in seconds to wait for function result to become truthy.
+        check_interval: Time in seconds to wait between ready checks.
+
+    Raises:
+        TimeoutError: if the callback function did not return a truthy value within timeout.
+    """
+    start_time = now = datetime.now()
+    min_wait_seconds = timedelta(seconds=timeout)
+    while now - start_time < min_wait_seconds:
```

Comment:
Maybe you can change this to `while datetime.now() - start_time < min_wait_seconds:` to simplify the code a bit.

Reply:
I think it's a tradeoff between line 157's `start_time = now = datetime.now()` vs `while datetime.now() - start_time < min_wait_seconds`, so either seems fine :D